support/historyarchive/xdrstream.go

```diff
@@ -21,25 +21,48 @@ import (
 
 type XdrStream struct {
 	buf        bytes.Buffer
-	rdr        io.ReadCloser
+	rdr        *countReader
 	rdr2       io.ReadCloser
 	sha256Hash hash.Hash
 
 	validateHash bool
 	expectedHash [sha256.Size]byte
 }
 
+type countReader struct {
```

Comment:
I think it's worth preserving the full name of the type being wrapped, so that it is clear this is a `ReadCloser`.
```suggestion
type countReadCloser struct {
```

Reply:
is there a convention for naming structs which implement `io.ReadCloser` with a `ReadCloser` suffix? I found these examples from the standard library which just have a `Reader` suffix:

https://golang.org/pkg/io/#PipeReader
https://golang.org/pkg/compress/gzip/#NewReader

Reply:
An nice, I didn't know that was a common pattern, TIL! 👍